app/main.py

```diff
@@ -1,25 +1,66 @@
 from abc import ABC
+from typing import Callable
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, obj: Callable, objtype: int = None) -> int:
+        value = getattr(obj, self.private_name)
+        return value
+
+    def __set__(self, obj: Callable, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(obj, self.private_name, value)
+        else:
+            raise ValueError
+
+    def __set_name__(self, owner: Callable, name: str) -> None:
+        self.public_name = name
+        self.private_name = "_" + name
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: SlideLimitationValidator
```

Comment:
In this case you are providing a class as limitation class, so the typehint would be: 
```suggestion
            limitation_class: type(SlideLimitationValidator)
```